app/javascript/controllers/file_auth_controller.js

```diff
@@ -32,18 +32,28 @@ export default class extends Controller {
     const manifest = evt.detail
     const canvases = manifest.items
     this.addPostCallbackListener()
-    const resources = canvases.flatMap((canvas) => {
+    this.documents = canvases.flatMap((canvas) => {
       const annotationPages = canvas.items
       return annotationPages.flatMap((annotationPage) => {
         const paintingAnnotations = annotationPage.items.filter((annotation) => annotation.motivation === "painting")
         return paintingAnnotations.map((annotation) => {
           const contentResource = annotation.body
-          this.maybeDrawContentResource(contentResource)
           return contentResource
         })
       })
     })
-    this.firstFile = resources[0]['id']
+    console.log("Resources: ", this.documents)
+    this.maybeDrawContentResource(this.documents[0]) // cause login to first resource
+    this.firstFile = this.documents[0]['id']
```

Comment:
I am not sure the firstFile is needed anymore. This was for the line ```if (this.resources[messageId].contentResourceId !== this.firstFile) return``` in loginNeeded. This was needed because this used to loop through every file on render and authorize them. since we are no longer doing this I don't think we need this or the line if (this.resources[messageId].contentResourceId !== this.firstFile) return.

Reply:
You are right. I removed it.